From b701673e7429336248c307c93c2c26f443719255 Mon Sep 17 00:00:00 2001
From: Bernd Schubert <bernd.schubert@fastmail.fm>
Date: Sun, 5 May 2024 13:09:56 +0200
Subject: [PATCH] Fix missing fuse_loop_cfg_destroy() in
 fuse_session_loop_mt_31 (#944)

All credits to Miklos Szeredi <miklos@szeredi.hu> for spotting
this.

Signed-off-by: Bernd Schubert <bernd.schubert@fastmail.fm>
Signed-off-by: Pavel Reichl <preichl@redhat.com>
---
 lib/fuse_loop_mt.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/fuse_loop_mt.c b/lib/fuse_loop_mt.c
index 70ff8f8..ecf8af8 100644
--- a/lib/fuse_loop_mt.c
+++ b/lib/fuse_loop_mt.c
@@ -440,10 +440,15 @@ int fuse_session_loop_mt_31(struct fuse_session *se, int clone_fd);
 FUSE_SYMVER("fuse_session_loop_mt_31", "fuse_session_loop_mt@FUSE_3.0")
 int fuse_session_loop_mt_31(struct fuse_session *se, int clone_fd)
 {
+	int err;
 	struct fuse_loop_config *config = fuse_loop_cfg_create();
 	if (clone_fd > 0)
 		 fuse_loop_cfg_set_clone_fd(config, clone_fd);
-	return fuse_session_loop_mt_312(se, config);
+	err = fuse_session_loop_mt_312(se, config);
+
+	fuse_loop_cfg_destroy(config);
+
+	return err;
 }
 
 struct fuse_loop_config *fuse_loop_cfg_create(void)
-- 
2.45.2

